---
title: "qa2"
output: html_document
---

```{R}
library(stats)
library(shiny)
library(shinyFiles)
library(ShortRead)
library(shinyjs)
library(dplyr)
library(ggplot2)

arquivo_fastq <- "C:/Users/vitor/Documents/ufpr/projeto-bioinformatica/SRR27316221/SRR27316221.fastq"
src <- QAFastqSource(con = arquivo_fastq)

qa_elements <- QACollate(
  src,
  QAReadQuality()
)

qa_elements <- QACollate(
  src,
  QAReadQuality(),
  QAAdapterContamination(Lpattern = "AGATCGGAAGAGCACACGTCTGAACTCCAGTCA"),  # adapter example
  QANucleotideUse(),
  QAQualityUse(),
  QASequenceUse(),
  QAFrequentSequence(n = 10, reportSequences = TRUE),
  QANucleotideByCycle(),
  QAQualityByCycle()
)


qa_result <- qa2(src, qa_elements, verbose = TRUE)
report(qa_result)

res <- qa2(FastqSampler(arquivo_fastq), QAReadQuality())
res <- qa2(FastqSampler(src@con), src@data, verbose = TRUE)
```

```{R}
library(Biostrings)
library(ShortRead)

adapters <- readDNAStringSet("path/to/adapters.fa")
names(adapters)  # view adapter names



fq <- readFastq("path/to/your.fastq")
seqs <- sread(fq)  # DNAStringSet

results <- sapply(adapters, function(adapter) {
  counts <- vcountPattern(adapter, seqs, max.mismatch = 2)
  sum(counts > 0)  # number of reads with at least one match
})

# Combine with names
data.frame(
  Adapter = names(results),
  ReadsWithAdapter = results,
  Percent = 100 * results / length(seqs)
)
```
